nixos/tests/paperless.nix

```diff
@@ -26,6 +26,9 @@ import ./make-test-python.nix ({ lib, ... }: {
         # Wait until server accepts connections
         machine.wait_until_succeeds("curl -fs localhost:28981")
 
+    with subtest("Task-queue gets ready"):
```

Comment:
If subtest `Add a document via the web interface` requires `paperless-task-queue.service`, better fix this in the service definition instead by adding `systemd.services.paperless-web.after = [ "paperless-task-queue.service" ]`.

Reply:
The `paperless-web.service` is started by `paperless-scheduler.service`. Independently, `paperless-consumer.service` and `paperless-task-queue.service` are also started by `paperless-scheduler.service`. 
All of them are needed for the consumption by the web-ui. I'm not too sure whether adding `paperless-task-queue.service` to `after` of `paperless-web.service` is needed here.

I guess an argument would be that `paperless-web.service` will fail to start, if the consumption pipeline is not up and running..

Reply:
> All of them are needed for the consumption by the web-ui. 

Then it makes sense to encode this in the webinterface service definition.
This ensures that the webinterface is fully functional when the service is active.

Reply:
I agree. But isn't that already the case? 
From systemd.services.paperless-web:
`after = [ "paperless-scheduler.service" ];`

Or shall we add a `required` to the web-ui systemd service?

Reply:
I'd also be fine with adding this comment:
```nix
    # Required for consuming documents via the web interface
    with subtest("Task-queue gets ready"):
```

Reply:
`paperless-scheduler` running doesn't imply that `paperless-task-queue` is running.
Yes, the correct way to add this dependency is setting `after` and `required`.

Reply:
But maybe it's better to keep both services separated.
- Simpler code
- The webinterface stays functional on task-queue errors

I'll let you decide.

Reply:
I just thought about it a bit more. There is an edge case, where the web-ui can function without (any) consumer: 
If just reading and editing the metadata of files, but without consuming of new files.
When we add `requires = [ "paperless-task-queue.service" "paperless-consumer.service" "paperless-scheduler"] ` to `systemd.services.paperless-web` we force the `web`unit to have those available, or fail.

Are we fine with that?

Reply:
For this reason I'd be fine with just https://github.com/NixOS/nixpkgs/pull/208723#discussion_r1060054304.